packages/cli/src/lib/Compiler.ts

```diff
@@ -369,7 +369,7 @@ export class Compiler {
       });
 
       const info: WrapManifest = {
-        abi: filteredAbi,
+        abi: (filteredAbi as unknown) as WrapAbi,
```

Comment:
This is a bit sketchy, and points out that we have 2 ABI types defined in `schema/parse/abi` and `manifest/wrap.info/abi`.

I looked into using the `manifest/wrap.info/abi` inside of the schema pipeline directly, but it is a bit complicated... I think we should create a new PR after this is merged to consolidate these definitions and only use the manifest's ABI.

Reply:
I added a final validation pass below this to ensure we don't write invalid ABIs.